### ISSUE TITLE
general search returns mails without the search term

### ISSUE THREAD

**Issue by ferdiga** [nextcloud/mail#9171]
### Steps to reproduce

1. install and configure mail 3.5
2. send en email to the address
3. search for terms which are not in the email header or body
4. random emails show up. (probably "important" ones
5. files with the search term in names are shown

### Expected behavior

only mails with the search term should be displayed.

### Actual behavior

show only mails with search term

### Mail app version

5.3

### Mailserver or service

world4you and others

### PHP engine version

PHP 8.2

### Web server

None

### Database

MySQL

### Additional info

running on Hetzner service NC 27.1.2

**Comment by GretaD** [comment]
I cannot reproduce this, but i only tested it on nc 28.0.0

**Comment by ferdiga** [comment]
so Ill have to wait until the upgrade is done by the provider

**Comment by GretaD** [comment]
> so Ill have to wait until the upgrade is done by the provider

no, if its an issue on 27, it has to be fixed, i just cannot test it on 27th because i have no 27th env. So my comment was that maybe the issue its not present on 28th.

**Comment by handkerchief333** [comment]
Same issues with 27.1.4, mail version 3.5.6, the search is rarely as one would wish. Wrong messages are displayed, or none at all, very inconsistent results and the search trigger is also buggy. Can't be more specific, just use the search itself and you'll see what I mean.

**Comment by handkerchief333** [comment]
I've updated, same results with 28.0.2, very buggy

**Comment by ferdiga** [comment]
@handkerchief333 
Did you check the normal (full text) search? I do not use it because I think the results are not reliable and the UI is bot usable as file names are cut off.
